### ISSUE TITLE
Import and evaluate a decision from another DMN

### ISSUE THREAD

**Issue by saig0** [camunda-community-hub/dmn-scala#71]
I can import other decisions from another DMN file. I can evaluate these decisions with the given import name as a prefix.

**Comment by falko** [comment]
Interesting feature.

**Comment by saig0** [comment]
**Idea:**
* two APIs: 
  * stateless: `parse(dmn)` + `eval(parsed-dmn)`
  * stateful: `deploy(dmn)` + `eval()`  
  * including Java APIs
* new dependency for the engine: `decision-repository`
  * `deploy()` + `findByName()` + `findById()` + `findAll()` + `remove()`
  * default implementation: `in-memory-decision-repository`

**Comment by falko** [comment]
The primary concepts that DMN offers for calling imported elements are BKM and Decision Service invocations.

**Comment by falko** [comment]
There should be examples in the [machine-readable files included with the spec](https://github.com/omg-dmn-taskforce/omg-dmn-spec/tree/master/examples) as well as in the [DMN TCK](https://github.com/dmn-tck/tck/tree/master/TestCases). If not, those would be great test cases to add ;-)

**Comment by saig0** [comment]
Yes, the DMN TCK has two test cases. Both are linked with this issue.

**Comment by saig0** [comment]
There is an open contribution to implement this feature: #215.

I also drafted a PoC to show how the feature can be implemented but with a different approach: #222.
